lib/D2L.Extensibility.AuthSdk.UnitTests/FunctionalApiUriTests.cs

```diff
@@ -288,6 +288,32 @@ public class FunctionalApiUriTests {
 		}
 
 		[Test]
+		public void UserContext_CreateAuthUri_QuerystringIsMaintained() {
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+
+			Assert.IsTrue(
+				queryUrl.ToString().Contains( TestConstants.API_PATH_QUERYSTRING ),
+				"The querystring parameter was not present"
+				);
+		}
+
+		[Test]
+		public void UserContext_CreateAuthUri_MultipleQuerystringsAreNotPresent() {
+			const string querystringCharacterEncoded = "%3F";
+
+			Uri queryUrl = m_userContext.CreateAuthenticatedUri( TestConstants.API_PATH_WITH_QUERY, "GET" );
+			bool duplicatedQuery = (

```

Comment:
The test does fail without the fix in place. Without the fix you can get "http://someurl%3Fquery=value&x_a...". We found this when updating the Valence C# SDK library in our web service test automation where it returned a SEE OTHER response because of the malformed URL. 

I wrote the test first, watched it fail, then added the fix. I'll update the name to end with "MultipleQuerystringSegmentsAreNotPresent".

In contrast, the other method that takes in a URI works. Because the string path is converted to a URL, this is where the problem is introduced.

I hope that helps!